pageserver/src/thread_mgr.rs

```diff
@@ -91,7 +91,7 @@ pub enum ThreadKind {
     // associated with one later, after receiving a command from the client.
     PageRequestHandler,
 
-    // Thread that connects to a safekeeper to fetch WAL for one timeline.
+    // Main walreciver thread that ensures that every timeline spanws a connection to safekeeper for WAL fecthing.
```

Comment:
Yes, the role had changed, I've adjusted the docs around, but the situation is so now, that
* there's an only thread, the main one that looks for more events from `tenant_mgr` and the cancellation
* there's a bunch of tokio tasks, 2 per timeline, that are launched for watching etcd and WAL streaming correspondingly

Nothing else, besides some extra threads that are used during `tokio::spawn` (none now, since we use the one threaded runtime, but can be more if we switch to another flavour) and extra threads that are used during `tokio::task::spawn_blocking` (~2 per timeline init, short-lived)